lib/rotoscope.rb

```diff
@@ -34,17 +34,16 @@ def self.trace(dest, blacklist: [], flatten: false)
   end
 
   def flatten(dest)
-    io_given = false
-    dest_file = if dest.respond_to?(:to_io)
-      io_given = true
-      dest.to_io
+    io_given = if dest.respond_to?(:puts)
```

Comment:
Well, it looks like Rotoscope.trace is supposed to support an IO object, but will have a similar problem if Zlib::GzipWriter is given unless it also provides `flatten: true`.  Since the rotoscope C extension uses `fprintf` to write to the output, it won't work directly with an IO object.